tests/infra_test.go

```diff
@@ -975,3 +971,34 @@ func getKeysFromMetrics(metrics map[string]float64) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+func checkError(err error) error {
+	var e *neturl.Error
+	if errors.As(err, &e) && e.Timeout() {
+		return nil
+	} else {
+		return err
+	}
+}
+
+func waitHttpResponse(errorsChan chan error, concurrency int) (int, error) {
+	expectedErrorsCount := 0
+	var err error
+	for ix := 0; ix < concurrency; ix++ {
+		select {
+		case err = <-errorsChan:
+			if err != nil {
+				err = checkError(err)
+				if err != nil {
+					break
+				}
+				expectedErrorsCount += 1
+			}
+		case <-time.After(5 * time.Second):
```

Comment:
the serial time will happen at most one time in case of a failure, the select will catch the first one of those events, so it doesnt had time, unless a fatal which shouldnt occur happen and then the test fails (for example hanged producer).
anyway i will remove it, even that i dont agree that its not enough.
it makes the consumer  / producer each one protected on its own without dependency.

Reply:
Removed the timeout, as its also break the draining (one would not been consumed) so if we remove the channel buffer we must remove the timeout as well.